src/eddymotion/model/base.py

```diff
@@ -521,6 +522,72 @@ class DKIModel(BaseDWIModel):
     _model_class = "dipy.reconst.dki.DiffusionKurtosisModel"
 
 
+class GaussianProcessModel(BaseModel):
+    """A Gaussian process model for DWI data based on [Andersson15]_."""
+
+    __slots__ = (
+        "_dwi",
+        "_kernel",
+        "_gpr",
+    )
+
+    def __init__(self, dwi, kernel, **kwargs):
+        """Implement object initialization.
+
+        Parameters
+        ----------
+        dwi : :obj:`~eddymotion.dmri.DWI`
+            The DWI data.
+        kernel : :obj:`~sklearn.gaussian_process.kernels.Kernel`
+            Kernel instance.
+        """
+
+        self._dwi = dwi
```

Comment:
```suggestion
        self._gtab = _rasb2dipy(gtab)
```

Reply:
This is now deprecated after https://github.com/jhlegarreta/eddymotion/pull/7, neither the signal nor the gradient table are slots of the class in the new signature, so resolving. We'll see how the new framework comes into life after we stabilize things.

Reply:
This is now deprecated after https://github.com/jhlegarreta/eddymotion/pull/7, neither the signal nor the gradient table are provided when initializing the class in the new signature, so resolving. We'll see how the new framework comes into life after we stabilize things.